### PR TITLE
feat: add OtcMakerOrder type into LimitOrderBook

### PR DESCRIPTION
**Pull request by DylanYang0523**
### PR Reminders
---

1. **Contract**: make sure the code follows our convention, ex: naming, explicit returns, etc.; if uncertain, discuss with others

2. **Test**: make sure tests can cover most normal and edge cases; if not, open follow-up tickets. Also, look out for failed tests and fix them!

3. **CHANGELOG.md**: update when external interfaces are changed

4. Workflow: 
    - Github: assign the pr to yourself; if pairing, can merge directly; else, assign someone to help review
    - Asana: assign the corresponding ticket to yourself and leave the pr link on it for easier follow-ups
    - Discord: if someone is mentioned in this pr, tag on Discord

### CODE REVIEW

**Comment by DylanYang0523** [comment]
Need to fix the test "PerpPortal test # getLiquidationPrice # liquidation price is correct when short with 5x leverage".

File: contracts/interface/IOtcMaker.sol [hunk 1]
```diff
@@ -13,22 +13,56 @@ interface IOtcMaker is IOtcMakerStruct, IOtcMakerEvent {
     // EXTERNAL NON-VIEW
     //
 
+    /// @notice Opens a position for trader by the given limit order,
+    ///         using the specified JIT liquidity parameters and signature.
+    /// @dev This function can only be called by the designated caller.
+    /// @param limitOrderParams The parameters of the limit order.
+    /// @param jitLiquidityParams The JIT liquidity parameters.
+    /// @param signature The signature associated with the limit order.
+    /// @return response The response containing the exchanged position size and notional value.
```
Comment (tailingchen) [inline contracts/interface/IOtcMaker.sol:22]: Rephrase: 
```
The response includes the exchanged position size and notional on OTCMaker.
```

File: contracts/interface/IOtcMaker.sol [hunk 2]
```diff
@@ -13,22 +13,56 @@ interface IOtcMaker is IOtcMakerStruct, IOtcMakerEvent {
...
         JitLiquidityParams calldata jitLiquidityParams,
         bytes calldata signature
     ) external returns (OpenPositionForResponse memory);
 
+    /// @notice Opens a position in the clearing house using the provided parameters.
+    /// @dev This function can only be called by the designated position manager.
+    /// @param params The parameters for opening the position.
+    /// @return base The amount of baseToken the taker got or spent
+    /// @return quote The amount of quoteToken the taker got or spent
+    /// @dev Delegates the call to the clearing house to open the position.
```
Comment (tailingchen) [inline contracts/interface/IOtcMaker.sol:34]: Arrange the tags on the document in sequential order.

File: contracts/interface/IOtcMaker.sol [hunk 3]
```diff
@@ -13,22 +13,56 @@ interface IOtcMaker is IOtcMakerStruct, IOtcMakerEvent {
...
+    /// @dev This function can only be called by the designated fund owner.
+    /// @param token The address of the token to withdraw.
+    /// @param amount The amount of tokens to withdraw.
+    /// @dev Delegates the call to the vault to withdraw the tokens and transfers them to the fund owner.
     function withdraw(address token, uint256 amount) external;
 
+    /// @notice Withdraws the entire balance of a specific token from the contract and transfers it to the fund owner.
+    /// @dev This function can only be called by the designated fund owner.
+    /// @param token The address of the token to withdraw.
+    /// @dev Retrieves the balance of the specified token and transfers the full amount to the fund owner.
```
Comment (tailingchen) [inline contracts/interface/IOtcMaker.sol:55]: mv to after L53

File: contracts/interface/IOtcMaker.sol [hunk 4]
```diff
@@ -13,22 +13,56 @@ interface IOtcMaker is IOtcMakerStruct, IOtcMakerEvent {
...
     function withdrawToken(address token) external;
 
+    /// @notice Claims a specific week of rewards for a liquidity provider using Merkle proofs.
+    /// @dev This function can only be called by the designated fund owner.
+    /// @param merkleRedeem The address of the MerkleRedeem contract.
+    /// @param liquidityProvider The address of the liquidity provider.
+    /// @param week The week number to claim rewards for.
+    /// @param claimedBalance The claimed amount.
+    /// @param _merkleProof The Merkle proofs for the verification.
+    /// @dev Delegates the call to the MerkleRedeem contract to claim rewards for the liquidity provider.
```
Comment (tailingchen) [inline contracts/interface/IOtcMaker.sol:65]: mv to after L59

File: contracts/interface/IOtcMaker.sol [hunk 5]
```diff
@@ -13,22 +13,56 @@ interface IOtcMaker is IOtcMakerStruct, IOtcMakerEvent {
...
+    /// @param jitLiquidityParams The JIT liquidity parameters.
+    /// @param signature The signature associated with the limit order.
+    /// @return The response includes the exchanged position size and notional on OTCMaker.
     function openPositionFor(
         ILimitOrderBook.LimitOrder calldata limitOrderParams,
         JitLiquidityParams calldata jitLiquidityParams,
         bytes calldata signature
-    ) external returns (int256 exchangedPositionSize, int256 exchangedPositionNotional);
-
+    ) external returns (int256, int256);
```
Comment (tailingchen) [inline contracts/interface/IOtcMaker.sol:27]: Suggest to keep named return here and fix natspec.

**Review by tailingchen — APPROVED** [review]
